src/ir/tag_claim_operation.h

```diff
@@ -0,0 +1,47 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_TAG_CLAIM_OPERATION_H_
+#define SRC_IR_TAG_CLAIM_OPERATION_H_
+
+#include <vector>
+
+#include "absl/strings/string_view.h"
+#include "src/ir/access_path_v2.h"
+#include "src/ir/data_decl.h"
+#include "src/ir/operation.h"
+
+namespace raksha::ir {
+
+// Description of an operation in the IR.
+class TagClaimOperation : public Operation {
```

Comment:
I think I don't understand what this class is for. Is this meant to be like `PrinX says DataX hasTag TagX` (which can't quite be right because not all the elements needed for that are here). If this is meant to be for adding tags to locations that store data, another thing that confuses me is that I thought `Operation`s are the actual functions that are applied to the data, whereas tags are more like an abstraction of the data. So this seems like it would be the specification/abstraction bleeding into the implementation.

Reply:
I did not want to operations to be used for only modeling concrete operations. It could be used to model concrete operations as well.